core/src/datayoga_core/blocks/relational/write/block.py

```diff
@@ -81,6 +82,18 @@ async def run(self, data: List[Dict[str, Any]]) -> BlockResult:
             self.execute(self.tbl.insert(), data)
             return utils.all_success(data)
 
+    def explode_if_specified(self, records: List[Dict[str, Any]]) -> List[Dict[str, Any]]:
+        """Explodes the input records into multiple records if the `foreach` flag is set to true,
+            and returns the original records as is otherwise.
+
+        Args:
+            records (List[Dict[str, Any]]): The input records to process
+
+        Returns:
+            List[Dict[str, Any]]: The processed records
+        """
+        return utils.explode_records(records, self.foreach) if self.foreach else records
```

Comment:
I would do the "if" outside of the method and only call it if needed. seems strange to have an "if_specified" method